src/test/java/org/jenkinsci/plugins/saml/LiveTest.java

```diff
@@ -0,0 +1,215 @@
+    }
+    */
+
+    private String readIdPMetadataFromURL() throws IOException {
+        // get saml metadata from IdP
+        URL metadata = new URL(createIdPMetadataURL());
+        URLConnection connection = metadata.openConnection();
+        return IOUtils.toString(connection.getInputStream(), StandardCharsets.UTF_8);
+    }
+
+    private String createIdPMetadataURL() {
+        return "http://" + samlContainer.getHost() + ":" + samlContainer.getFirstMappedPort() + "/simplesaml/saml2/idp/metadata.php";
+    }
+
+    private static void configureAuthorization() {
+        Jenkins.get().setAuthorizationStrategy(new MockAuthorizationStrategy().
+                grant(Jenkins.ADMINISTER).everywhere().to("group1").
+                grant(Jenkins.READ).everywhere().to("group2"));
+    }
+
+    private static SamlSecurityRealm configureBasicSettings(IdpMetadataConfiguration idpMetadataConfiguration, SamlAdvancedConfiguration advancedConfiguration) throws IOException {
+        // TODO use @DataBoundSetter wherever possible and load defaults from DescriptorImpl
+        File samlKey = new File(Jenkins.get().getRootDir(), "saml-key.jks");
+        FileUtils.copyURLToFile(LiveTest.class.getResource("LiveTest/saml-key.jks"), samlKey);
+        SamlEncryptionData samlEncryptionData = new SamlEncryptionData(samlKey.getAbsolutePath(), Secret.fromString("changeit"), Secret.fromString("changeit"), null, false);
+        return new SamlSecurityRealm(idpMetadataConfiguration, "displayName", "eduPersonAffiliation", 86400, "uid", "email", null, advancedConfiguration, samlEncryptionData, "none", SAML2_REDIRECT_BINDING_URI, Collections.emptyList());
+    }
+
+    private void startSimpleSAML(String rootUrl) throws IOException, InterruptedException {
+        samlContainer.
+                withEnv("SIMPLESAMLPHP_SP_ENTITY_ID", SERVICE_PROVIDER_ID).
+                withEnv("SIMPLESAMLPHP_SP_ASSERTION_CONSUMER_SERVICE", rootUrl + "securityRealm/finishLogin"). // login back URL
+                withEnv("SIMPLESAMLPHP_SP_SINGLE_LOGOUT_SERVICE", rootUrl + "logout"); // unused
+        samlContainer.start();
+        samlContainer.copyFileToContainer(MountableFile.forClasspathResource("org/jenkinsci/plugins/saml/LiveTest/users.php"), "/var/www/simplesamlphp/config/authsources.php"); // users info
+        samlContainer.copyFileToContainer(MountableFile.forClasspathResource("org/jenkinsci/plugins/saml/LiveTest/config.php"), "/var/www/simplesamlphp/config/config.php"); // config info,
+        samlContainer.copyFileToContainer(MountableFile.forClasspathResource("org/jenkinsci/plugins/saml/LiveTest/saml20-idp-hosted.php"), "/var/www/simplesamlphp/metadata/saml20-idp-hosted.php"); //IdP advanced configuration
+    }
+
+    private static void makeLoginWithUser1(JenkinsRule r) throws Exception {
+        // TODO commenceLogin fails with: SAMLException: Identity provider has no single sign on service available for the selected profileHTTP-Redirect
```

Comment:
I have no idea what to do with this. AFAICT the Docker container has the right stuff AFAICT and the Jenkins `config.xml` looks like I guess it should (knowing _nothing_ about SAML).

Reply:
I think this error was resolved—mismatch between display name and actual name of a selection box. Now I am getting some other error…